### PR TITLE
[Merged by Bors] - feat(Combinatorics/SimpleGraph): distance of vertices in a subgraph

### PR DESCRIPTION
**Pull request by Rida-Hamadani**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/SimpleGraph/Metric.lean [hunk 1]
```diff
@@ -234,6 +234,13 @@ lemma Connected.exists_path_of_dist (hconn : G.Connected) (u v : V) :
   obtain ⟨p, h⟩ := hconn.exists_walk_length_eq_dist  u v
   exact ⟨p, p.isPath_of_length_eq_dist h, h⟩
 
+/-- Supergraphs have smaller or equal distances to their subgraphs. -/
+theorem dist_le_subgraph_dist {G' : SimpleGraph V} {u v : V} (h : G ≤ G') (hd : G.dist u v ≠ 0) :
```
Comment (b-mehta) [inline Mathlib/Combinatorics/SimpleGraph/Metric.lean:238]: edist version?

File: Mathlib/Combinatorics/SimpleGraph/Metric.lean [hunk 2]
```diff
@@ -128,6 +128,13 @@ theorem edist_eq_one_iff_adj : G.edist u v = 1 ↔ G.Adj u v := by
     exact w.adj_of_length_eq_one <| Nat.cast_eq_one.mp <| h ▸ hw
   · exact le_antisymm (edist_le h.toWalk) (ENat.one_le_iff_pos.mpr <| edist_pos_of_ne h.ne)
 
+/-- Supergraphs have smaller or equal extended distances to their subgraphs. -/
+theorem edist_le_subgraph_edist {G' : SimpleGraph V} {u v : V} (h : G ≤ G') (hr : G.Reachable u v) :
```
Comment (IvanRenison) [inline Mathlib/Combinatorics/SimpleGraph/Metric.lean:132]:
```suggestion
theorem edist_le_subgraph_edist {G' : SimpleGraph V} (h : G ≤ G') (hr : G.Reachable u v) :
```

File: Mathlib/Combinatorics/SimpleGraph/Metric.lean [hunk 3]
```diff
@@ -234,6 +241,13 @@ lemma Connected.exists_path_of_dist (hconn : G.Connected) (u v : V) :
   obtain ⟨p, h⟩ := hconn.exists_walk_length_eq_dist  u v
   exact ⟨p, p.isPath_of_length_eq_dist h, h⟩
 
+/-- Supergraphs have smaller or equal distances to their subgraphs. -/
+theorem dist_le_subgraph_dist {G' : SimpleGraph V} {u v : V} (h : G ≤ G') (hr : G.Reachable u v) :
```
Comment (IvanRenison) [inline Mathlib/Combinatorics/SimpleGraph/Metric.lean:245]:
```suggestion
theorem dist_le_subgraph_dist {G' : SimpleGraph V} (h : G ≤ G') (hr : G.Reachable u v) :
```

File: Mathlib/Combinatorics/SimpleGraph/Metric.lean (hunk 2 above)
Comment (b-mehta) [inline Mathlib/Combinatorics/SimpleGraph/Metric.lean:132]: This ought to be provable without assuming reachability, right? If u isn't reachable from v in G then the rhs is infinity

**Review by b-mehta — APPROVED** [review]
bors d+

File: Mathlib/Combinatorics/SimpleGraph/Metric.lean (hunk 2 above)
Comment (Rida-Hamadani) [reply Mathlib/Combinatorics/SimpleGraph/Metric.lean]: Thank you, I overlooked that.

**Pull request closed without merging**